### PR TITLE
Enforce limit on FTE max partition count

### PR DESCRIPTION
**Pull request by linzebing**
## Description

To ensure the session property isn't accidentally set to an overwhelming number.

## Additional context and related issues

N/A

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/SystemSessionProperties.java [hunk 1]
```diff
@@ -1400,8 +1400,12 @@ private static Integer validateIntegerValue(Object value, String property, int l
 
         int intValue = (int) value;
         if (intValue < lowerBoundIncluded) {
-            throw new TrinoException(INVALID_SESSION_PROPERTY, format("%s must be equal or greater than %s", property, lowerBoundIncluded));
+            throw new TrinoException(INVALID_SESSION_PROPERTY, format("%s must be equal or greater than %d", property, lowerBoundIncluded));
```
Comment (arhimondr) [inline core/trino-main/src/main/java/io/trino/SystemSessionProperties.java:1403]: nit: we usually print integers with `%s`, don't remember the reason though

File: core/trino-main/src/main/java/io/trino/SystemSessionProperties.java [hunk 2]
```diff
@@ -264,19 +264,19 @@ public SystemSessionProperties(
                         MAX_HASH_PARTITION_COUNT,
                         "Maximum number of partitions for distributed joins and aggregations",
                         queryManagerConfig.getMaxHashPartitionCount(),
-                        value -> validateIntegerValue(value, MAX_HASH_PARTITION_COUNT, 1, false),
+                        value -> validateIntegerValue(value, MAX_HASH_PARTITION_COUNT, 1, Integer.MAX_VALUE, false),
```
Comment (arhimondr) [inline core/trino-main/src/main/java/io/trino/SystemSessionProperties.java:267]: Have you considered an overload?

**Review by arhimondr — APPROVED** [review]
LGTM % questions

File: core/trino-main/src/main/java/io/trino/SystemSessionProperties.java [hunk 3]
```diff
@@ -918,19 +918,19 @@ public SystemSessionProperties(
                         FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT,
                         "Maximum number of partitions for distributed joins and aggregations executed with fault tolerant execution enabled",
                         queryManagerConfig.getFaultTolerantExecutionMaxPartitionCount(),
-                        value -> validateIntegerValue(value, FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT, 1, false),
+                        value -> validateIntegerValue(value, FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT, 1, 1000, false),
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/SystemSessionProperties.java:921]: nit: extract constant for 1000

Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/SystemSessionProperties.java:921]: Also I am not a big fan of hardcoding 1000. It is a bit arbitrary.
Probably it should be limitted to `queryManagerConfig.getFaultTolerantExecutionMaxPartitionCount()` and we would need an extra config parameter to set default value of session property.

But I do not feel to strongly about that.

File: core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java [hunk 4]
```diff
@@ -883,6 +883,7 @@ public QueryManagerConfig setFaultTolerantExecutionTaskDescriptorStorageMaxMemor
     }
 
     @Min(1)
+    @Max(1000)
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java:886]: constant (share with SystemSessionProperties)

**Review by losipiuk — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/SystemSessionProperties.java (hunk 3 above)
Comment (linzebing) [reply core/trino-main/src/main/java/io/trino/SystemSessionProperties.java]: It's weird to have too many configs:) I don't see any similar configs like that, so I extract constant (similar to `MAX_TASK_RETRY_ATTEMPTS`)

Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/SystemSessionProperties.java]: `MAX_TASK_RETRY_ATTEMPTS` is different animal. As with that you just set number of attempts. And here you specify `FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT` you specify number of partitions and `1000` hard coded limit on what you can set. There is no such hardcoded limit for `MAX_TASK_RETRY_ATTEMPTS`. 

Let's leave 1000 in the code for now. We can revisit later if needed.

**Review by losipiuk — APPROVED** [review]

**Pull request merged**